### ISSUE TITLE
VA_FORMAT_STRING_USES_NEWLINE - False positive in java 17 text block feature

### ISSUE THREAD

**Issue by zielinskikamil** [spotbugs/spotbugs#1877]
Hi there,
Seems that spotbugs report following error code VA_FORMAT_STRING_USES_NEWLINE on.
```
    // java 17
    public static final String ACCOUNT_QUERY = """
        SELECT id,
               name
        FROM   account
        WHERE  id = '%s'
        """;
    String.format(ACCOUNT_QUERY, "1");
```

AFAIK """ don't accept %n. Guys can you take a look at it? Java 17 is becoming more popular.

**Comment by UngenuineGuineapig** [comment]
+1, quite annoying

**Comment by ThrawnCA** [comment]
In the specific case of generating an SQL query from a format string, I imagine you don't need platform-specific line separators. It would be a good candidate for excluding the rule on the affected classes/methods.

**Comment by UngenuineGuineapig** [comment]
Actually, we found out yesterday that using 
`
"""
something something %s
""".formatted(varString)
`
does not create the warning. So I guess it's to using .formatted instead of String.format for now.

**Comment by gtoison** [comment]
TLDR: use `%n\` to get platform line endings with formatters and text blocks
This should be explained in the bug description with #2881
